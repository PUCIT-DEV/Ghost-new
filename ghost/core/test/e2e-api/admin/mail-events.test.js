const assert = require('assert');
const {agentProvider} = require('../../utils/e2e-framework');
const configUtils = require('../../utils/configUtils');
const mailEvents = require('../../../core/server/services/mail-events');
const models = require('../../../core/server/models');

describe('Mail Events API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
    });

    it('Can add a mail event', async function () {
        configUtils.set('hostSettings:siteId', 123);
        configUtils.set('hostSettings:mailEventsSecretKey', 'foobarbaz');

        // Re-initialise the mail events service with the new config values
        mailEvents.init();

        const payload = {
            // The signature is based on the previous two config values as well as the
            // "events" array below. If you change any of these values, you will need to
            // update the signature otherwise the request will fail
            signature: 'c505cb9b343795954199d5d514bf520b6b133bd9f1580805f0fa150d1b89fdab',
            events: [
                {
                    id: 'Ase7i2zsRYeDXztHGENqRA',
                    timestamp: 1521243339.873676,
                    event: 'opened',
                    message: {
                        headers: {
                            'message-id': '20130503182626.18666.16540@sandboxb052085d6a7b401bb117d3a432d1d659.mailgun.org'
                        }
                    },
                    recipient: 'alice@example.com'
                }
            ]
        };

        await agent
            .post('/mail_events/', {
                headers: {
                    'content-type': 'application/json'
                }
            })
            .body(payload)
            .expectStatus(200)
            .matchBodySnapshot({})
            .matchHeaderSnapshot({});

        const storedMailEvent = await models.MailEvent.findOne({id: 'Ase7i2zsRYeDXztHGENqRA'});

        assert.ok(storedMailEvent, 'Expected mail event was not found in the database');
    });
});
