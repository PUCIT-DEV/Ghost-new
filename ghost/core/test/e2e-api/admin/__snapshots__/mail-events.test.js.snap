// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mail Events API Can add a mail event 1: [body] 1`] = `Object {}`;

exports[`Mail Events API Can add a mail event 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "2",
  "content-type": "application/json; charset=utf-8",
  "content-version": "v5.51",
  "etag": "W/\\"2-vyGp6PvFo4RvsFtPoIWeCReyIC8\\"",
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
